### PR TITLE
FOSSA license scanning GitHub Action added

### PR DESCRIPTION
**Pull request by idvoretskyi**
Fixes https://github.com/kedacore/keda/issues/808

Signed-off-by: Ihor Dvoretskyi ihor@linux.com

### CODE REVIEW

**Comment by idvoretskyi** [comment]
FYI - to have this action work properly, we'll have to add the FOSSA API key as a GitHub secrets variable (`secrets.FOSSA_API_KEY`).

**Comment by zroubalik** [comment]
@idvoretskyi how can I get FOSSA API key? I can add it to Github secrets.

File: .github/workflows/fossa.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+      with:
+        go-version: '^1.13.1'
+    - run: go version
+    # Runs a set of commands to initialize and analyze with FOSSA
+    - name: Install and run FOSSA analysis
```
Comment (zroubalik) [inline .github/workflows/fossa.yml:19]: Might be worth to add fossa-cli to our build tools image: https://github.com/kedacore/keda/blob/389ea136d669e2f46cbc09b222d3255ebb0361aa/tools/build-tools.Dockerfile and here just run the analyze.
This way we have all tools consistently installed in one image.
WDYT?

Comment (idvoretskyi) [reply .github/workflows/fossa.yml:19]: @zroubalik np, let's do it :)

**Comment by zroubalik** [comment]
@ahmelsayed I hitting an issue when trying to build the tools image today (including the changes from this PR, but it is not relevant), I haven't seen such problem before. By chance do you have an idea what could go wrong? Is there any problem with the microsoft repo?
```
...
Creating config file /etc/apt/apt.conf.d/50unattended-upgrades with new version
Setting up python3-software-properties (0.96.24.32.12) ...
Setting up software-properties-common (0.96.24.32.12) ...
Processing triggers for libc-bin (2.27-3ubuntu1) ...
Processing triggers for dbus (1.12.2-1ubuntu1.1) ...
deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ bionic main
Warning: apt-key output should not be parsed (stdout is not a terminal)
Executing: /tmp/apt-key-gpghome.IKSNBCVfqT/gpg.1.sh --keyserver packages.microsoft.com --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
gpg: keyserver receive failed: Connection timed out
The command '/bin/sh -c apt-get install apt-transport-https lsb-release software-properties-common dirmngr -y &&     AZ_REPO=$(lsb_release -cs) &&     echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" |         tee /etc/apt/sources.list.d/azure-cli.list &&     apt-key --keyring /etc/apt/trusted.gpg.d/Microsoft.gpg adv         --keyserver packages.microsoft.com         --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF &&     apt-get update &&     apt-get install -y azure-cli' returned a non-zero code: 2
```

**Comment by idvoretskyi** [comment]
@zroubalik I've tried to build the local image using this [Dockerfile](https://github.com/kedacore/keda/blob/master/tools/build-tools.Dockerfile), had the same issue, so can confirm.

**Comment by ahmelsayed** [comment]
Yes, I'm seeing the same issue as well. Looks like a general issue with [that keyserver](https://github.com/kedacore/keda/blob/35b2ce8e9652e0a009221a97d6570b019cb914d9/tools/build-tools.Dockerfile#L13). 

```console
$ gpg  --keyserver packages.microsoft.com --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
gpg: keyserver receive failed: Connection timed out
```

changing it to the ubuntu keyserver worked. 

```console
$ gpg  --keyserver keyserver.ubuntu.com --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
gpg: key 0xEB3E94ADBE1229CF: "Microsoft (Release signing) <gpgsecurity@microsoft.com>" not changed
gpg: Total number processed: 1
gpg:              unchanged: 1
```
https://github.com/kedacore/keda/pull/861

File: .github/workflows/fossa.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: FOSSA
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```
Comment (ahmelsayed) [inline .github/workflows/fossa.yml:10]: I think you need to add 

```yaml
container: kedacore/build-tools:latest
```

to use the environment built from https://github.com/kedacore/keda/blob/master/tools/build-tools.Dockerfile

**Comment by zroubalik** [comment]
> Yes, I'm seeing the same issue as well. Looks like a general issue with [that keyserver](https://github.com/kedacore/keda/blob/35b2ce8e9652e0a009221a97d6570b019cb914d9/tools/build-tools.Dockerfile#L13).
> 
> ```
> $ gpg  --keyserver packages.microsoft.com --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
> gpg: keyserver receive failed: Connection timed out
> ```
> 
> changing it to the ubuntu keyserver worked.
> 
> ```
> $ gpg  --keyserver keyserver.ubuntu.com --recv-keys BC528686B50D79E339D3721CEB3E94ADBE1229CF
> gpg: key 0xEB3E94ADBE1229CF: "Microsoft (Release signing) <gpgsecurity@microsoft.com>" not changed
> gpg: Total number processed: 1
> gpg:              unchanged: 1
> ```
> 
> #861

Thanks, helped.

@idvoretskyi the image is updated.

File: .github/workflows/fossa.yml (hunk 2 above)
Comment (idvoretskyi) [reply .github/workflows/fossa.yml:10]: Ooops, thanks.

**Comment by idvoretskyi** [comment]
@zroubalik here we go ^

**Comment by idvoretskyi** [comment]
@zroubalik it seems that build fails because of the non-available FOSSA API key.

**Comment by ahmelsayed** [comment]
I think secrets are not passed to actions run from a fork.

**Comment by idvoretskyi** [comment]
@ahmelsayed they are not passed from the fork, but we have secrets added to the current repo itself.

**Comment by ahmelsayed** [comment]
I was referring to this note from github 

image

I'm thinking this is what's causing the setting to be empty in the workflow. I see @zroubalik added it 7 hours ago (though I can't see the value) 

I think if you remove the debug echo we should be able to merge it and see if it passes on master.

**Comment by zroubalik** [comment]
@ahmelsayed good catch, it seems like this could be the problem.

File: .github/workflows/fossa.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+          go-version: "^1.13.1"
+      - run: go version
+      # Runs a set of commands to initialize and analyze with FOSSA
+      - name: run FOSSA analysis
+        env:
+          FOSSA_API_KEY: ${{ secrets.FOSSA_API_KEY }}
+        run: |
+          export GOPATH=$HOME/go
+          export PATH=$PATH:$(go env GOPATH)/bin
+          echo $FOSSA_API_KEY
```
Comment (zroubalik) [inline .github/workflows/fossa.yml:26]: the echo is still here

Comment (idvoretskyi) [reply .github/workflows/fossa.yml]: @zroubalik done.

**Pull request merged**